src/components/functional/PaymentRequestDetailsModal/PaymentRequestDetailsModal.tsx

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useState } from 'react';
+import { useMerchantTags } from '../../../api/hooks/useMerchantTags';
+import UIPaymentRequestDetailsModal from '../../ui/PaymentRequestDetailsModal/PaymentRequestDetailsModal';
+import { remotePaymentRequestToLocalPaymentRequest, parseApiTagToLocalTag } from '../../../utils/parsers';
+import { LocalTag } from '../../../types/LocalTypes';
+import { PaymentRequestClient } from '../../../api/clients/PaymentRequestClient';
+import { PaymentRequest, PaymentRequestUpdate } from '../../../api/types/ApiResponses';
+import { MerchantClient } from '../../../api/clients/MerchantClient';
+
+interface PaymentRequestDetailsModalProps {
+  token: string; // Example: "eyJhbGciOiJIUz..."
+  apiUrl: string; // Example: "https://api.nofrixion.com/api/v1"
+  merchantId: string;
+  paymentRequestID: string;
+  open: boolean;
+  onDismiss: () => void;
+}
+const PaymentRequestDetailsModal = ({
+  token,
+  apiUrl,
+  merchantId,
+  paymentRequestID,
+  open,
+  onDismiss,
+}: PaymentRequestDetailsModalProps) => {
+  const paymentRequestClient = new PaymentRequestClient(apiUrl, token, merchantId);
+  const merchantClient = new MerchantClient(apiUrl, token, merchantId);
+  const merchantTags = useMerchantTags(apiUrl, token, merchantId);
+  const [localMerchantTags, setLocalMerchantTags] = useState<LocalTag[]>([] as LocalTag[]);
+  const [paymentRequest, setPaymentRequest] = useState<PaymentRequest | undefined>();
+
+  const onRefundClick = async (paymentAttemptID: string) => {
+    //TODO: Will implement refund for atleast card payment attempts. For PISP, it will need to be worked on later.
+    console.log(paymentAttemptID);
+  };
+
+  const onTagAdded = async (tag: LocalTag) => {
+    if (paymentRequest) {
+      const existingTagIds = paymentRequest.tags?.map((tag) => tag.id) ?? [];
+      const paymentRequestUpdate: PaymentRequestUpdate = {
+        tagIds: existingTagIds.concat(tag.id),
+      };
+      const paymentRequestTagAdd = await paymentRequestClient.update(paymentRequest.id, paymentRequestUpdate);
+      if (paymentRequestTagAdd.error) {
+        console.log(paymentRequestTagAdd.error);
+      } else {
+        setPaymentRequest(paymentRequestTagAdd.data);
+      }
+    }
+  };
+
+  const onTagCreated = async (tag: LocalTag) => {
```

Comment:
When tags are modified the payment request row doesn't update. It think this can be solved if the payment request is passed in by the PaymentRequestRow component.